File: python/first-non-repeating-character.py
#  Write a function that takes in a string of lowercase English-alphabet letters and returns the index of the string's first non-repeating character. The first non-repeating character is the first character in a string that occurs only once. If the input string doesn't have any non-repeating characters, your function should return -1

string = "abcdcaf"
# output = 1 b is the first character that only appears once in the string


def firstNonRepeatingCharacter(string):
    hash = {}
    for char in string:
        if char in hash:
            hash[char] += 1
        else:
            hash[char] = 1
    for key, value in hash.items():
        if value == 1:
            return string.index(key)
    return -1


print(firstNonRepeatingCharacter(string))

File: js/first-non-repeating-character.js
// Write a function that takes in a string of lowercase English-alphabet letters and returns the index of the string's first non-repeating character. The first non-repeating character is the first character in a string that occurs only once. If the input string doesn't have any non-repeating characters, your function should return -1

string = "abcdcaf";
// output = 1 // b is the first character that only appears once in the string

function firstNonRepeatingCharacter(string) {
    // Write your code here.
    let hash = {};
    for (let index = 0; index < string.length; index++) {
        if (string[index] in hash) {
            hash[string[index]] += 1;
        } else {
            hash[string[index]] = 1;
        }
    }
    for (let index = 0; index < string.length; index++) {
        if (hash[string[index]] === 1) {
            return index;
        }
    }
    return -1;
}
